src/main/java/cube/logic/command/PromotionCommand.java

```diff
@@ -54,6 +54,8 @@ public CommandResult execute(ModelManager model, StorageManager storage) throws
         FoodList list = model.getFoodList();
         PromotionList promotionList = model.getPromotionList();
         obtainPromotionFood(list);
+        CommandUtil.requirePromotionNotExists(promotionList, newPromotion.getName());
```

Comment:
It is correct to try to handle bugs related to repeated promotions, but this requirePromotionNotExists() method only checks for same food name. However, in real life situation, the same food can have different promotions at different time. Therefore, if this is changed to not allow overlapping time periods, it will be better.